### PR TITLE
feat: Add role prop to Text, View and Accessibility docs

### PR DESCRIPTION
**Pull request by gabrieldonadel**
This PR adds the `role` prop to the `Text`, `View` and `Accessibility` documentation

Related to https://github.com/facebook/react-native/commit/a50e6fb341eacd955071e4a7c6d734e7d3c94cd7, https://github.com/facebook/react-native/commit/20718e6b8ce4f5c3a8393067d5e8eb0da910751c and https://github.com/facebook/react-native/commit/e3e635ef84156a5283d378858f7c8deb44464fea

### CODE REVIEW

File: docs/accessibility.md [hunk 1]
```diff
@@ -348,6 +348,14 @@ Use this property to assign a custom function to be called when someone activate
...
+| enum(`'alert'`, `'alertdialog'`, `'application'`, `'article'`, `'banner'`, `'button'`, `'cell'`, `'checkbox'`, `'columnheader'`, `'combobox'`, `'complementary'`,`'contentinfo'`,`'definition'`, `'dialog'`, `'directory'`, `'document'`, `'feed'`, `'figure'`, `'form'`, `'grid'`, `'group'`, `'heading'`, `'img'`, `'link'`, `'list'`, `'listitem'`, `'log'`, `'main'`, `'marquee'`, `'math'`, `'menu'`, `'menubar'`, `'menuitem'`, `'meter'`, `'navigation'`, `'none'`, `'note'`, `'option'`, `'presentation'`, `'progressbar'`, `'radio'`, `'radiogroup'`, `'region'`, `'row'`, `'rowgroup'`, `'rowheader'`, `'scrollbar'`, `'searchbox'`, `'separator'`, `'slider'`, `'spinbutton'`, `'status'`, `'summary'`, `'switch'`, `'tab'`, `'table'`, `'tablist'`, `'tabpanel'`, `'term'`, `'timer'`, `'toolbar'`, `'tooltip'`, `'tree'`, `'treegrid'`, `'treeitem'`) |
```
[the remark this answers is not in the document]
Comment (gabrieldonadel) [reply docs/accessibility.md:357]: Sure, updated. 
image

**Comment by Simek** [comment]
Thank you for the updates! The idea behind requesting those changes was to have only one list of possible entries with descriptions, and refer to that list everywhere.

If you want to link "role" instead of "accessiblityRole", you should move possible roles the listing from "accessiblityRole" to "role", and in "accessiblityRole" link "Role" as type, like for Text or View pages.

Screenshot 2022-11-10 094947

**Comment by gabrieldonadel** [comment]
@Simek `Role` and `AccessibilityRole` support different values, e.g. `slider` only exists on `Role`, so we can't make everything point to `AccessibilityRole`

**Comment by Simek** [comment]
Yeah, unfortunately, I see that in the code:
* https://github.com/facebook/react-native/blob/main/Libraries/Components/View/ViewAccessibility.js

IMO, it would be nice to have a type which represents intersection of those both types and then extend this type by unique values for each of the props, at least for the transition period.

So, to summarize, this PR looks good. It would be awesome to replace the type table with a list of values with descriptions, like we have for `accessiblityRole`, but this also can be done in the separate PR in the future.

**Comment by gabrieldonadel** [comment]
> It would be awesome to replace the type table with a list of values with descriptions, like we have for `accessiblityRole`

Sure thing, I'll work on that then

**Comment by gabrieldonadel** [comment]
> Yeah, unfortunately, I see that in the code:
> 
> * https://github.com/facebook/react-native/blob/main/Libraries/Components/View/ViewAccessibility.js
> 
> IMO, it would be nice to have a type which represents intersection of those both types and then extend this type by unique values for each of the props, at least for the transition period.
> 
> So, to summarize, this PR looks good. It would be awesome to replace the type table with a list of values with descriptions, like we have for `accessiblityRole`, but this also can be done in the separate PR in the future.

@Simek, I need an opinion on this if you don't mind. Many of these options are simply mapped to `undefined` and will have no real impact on the user, should that be documented somehow? E.g. using `role={'banner'}` does absolutely anything. Show we even list these values?

**Comment by gabrieldonadel** [comment]
Hi @Simek I've updated the list to include a description for each value that is not mapped to `undefined`. I opted for not listing those as it add no real value to Android/iOS users

**Review by Simek — APPROVED** [review]
Thank you for all of the updates, this looks great! 👍

**Comment by Simek** [comment]
> E.g. using role={'banner'} does absolutely anything. Show we even list these values?

Make sense for now, I think it's better than listing those option as unimplemented, but we will need to revisit this list before next release to make sure that it is up to date with implemented roles.

**Pull request merged**